Sources/NIOHTTPCompression/HTTPResponseDecompressor.swift

```diff
@@ -0,0 +1,199 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CNIOExtrasZlib
+import NIO
+import NIOHTTP1
+
+public final class NIOHTTPResponseDecompressor: ChannelDuplexHandler, RemovableChannelHandler {
+    public typealias InboundIn = HTTPClientResponsePart
+    public typealias InboundOut = HTTPClientResponsePart
+    public typealias OutboundIn = HTTPClientRequestPart
+    public typealias OutboundOut = HTTPClientRequestPart
+
+    /// Specifies how to limit decompression inflation.
+    public enum DecompressionLimit {
```

Comment:
Should we make this a `struct DecompressionLimit` with an inner private `enum` so that we can extend it to have other limits later without breaking public API? CC @Lukasa

Reply:
refactored, thanks